cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,981 @@
+|    `cudf::table *`   |     `cudf::mutable_table_view`    |                                                 |
+| `cudf::device_table` |     `cudf::table_device_view`     |                    Immutable                    |
+| `cudf::device_table` | `cudf::mutable_table_device_view` |                     Mutable                     |
+| `gdf_scalar`         | `cudf::scalar`                    |                     Not yet implemented         |
+
+### Functions
+
+|              Old             |                       New                       |                                     Notes                                    |
+|:----------------------------:|:-----------------------------------------------:|:----------------------------------------------------------------------------:|
+|       `gdf_is_valid()`       |        `*column_device_view::is_valid()`        |         Verifies existence of bitmask before  attempting to read it.         |
+|    `bit_mask::is_valid()`    |    `*column_device_view::is_valid_nocheck()`    | Does not verify the existence of the bitmask  before attempting to read it.  |
+| `gdf_vald_allocation_size()` |        `bitmask_allocation_size_bytes()`        |                                                                              |
+|        `is_nullable()`       |               `*view::nullable()`               |                                                                              |
+|         `has_nulls()`        |               `*view::has_nulls()`              |                                                                              |
+|        `cudf::copy()`        | `column::column(const&)` `table::table(const&)` |                               Copy constructors                              |
+|    `cudf::allocate_like()`   |      `cudf::allocate_like()`      |                      This applies to both columns/tables                     |
+|     `cudf::empty_like()`     |        `cudf::empty_like()`       |                      This applies to both columns/tables                     |
+
+## Strings Support<a name="string_support"></a>
+
+One of the more significant changes in `libcudf++` is the addition of native support for columns of strings. 
+Previously, string support in libcudf was accomplished via the `NVCategory` class.
+See the [NVCategory](#nvcategory_changes) section for more detail. 
+This class will no longer exist in `libcudf++`, but a replacement "dictionary" column type is planned for the future to replace the functionality of NVCategory.
+
+All libcudf functions will now need to be able to operate on a column of string elements directly. 
+This poses a set of unique challenges, as unlike all previous libcudf types, the elements in a string column are variable width. 
+In order to represent variable width strings, libcudf uses a *compound* column (i.e., column with children).
```

Comment:
There should probably be a previous section generically describing column children. That can then lead into this section on strings. Then add a section on lists and structs.